core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -193,9 +196,11 @@ public SnapshotInfo snapshot(Snapshot snapshot) {
      * @param listener snapshot creation listener
      */
     public void createSnapshot(final SnapshotRequest request, final CreateSnapshotListener listener) {
-        final Snapshot snapshot = request.snapshot;
-        validate(snapshot);
-        final SnapshotId snapshotId = SnapshotId.createNew(snapshot);
+        final String repositoryName = request.repositoryName;
+        final String snapshotName = request.snapshotName;
+        validate(repositoryName, snapshotName);
+        final SnapshotId snapshotId = new SnapshotId(snapshotName); // creates a new UUID for the snapshot
```

Comment:
I prefer that we make this explicit by passing `UUIDs.randomBase64UUID()` here and removing the overloaded constructor.